dashboard/app/controllers/report_abuse_controller.rb

```diff
@@ -109,4 +84,121 @@ def report_abuse_form
+    end
+    render json: {abuse_score: value}
+  end
+
+  # DELETE /v3/channels/:channel_id/abuse
+  # Clear an abuse score. Requires project_validator permission
+  def reset_abuse
+    return head :unauthorized unless can?(:destroy_abuse, nil)
+
+    channel_id = params[:channel_id]
+
+    begin
+      value = Projects.new(get_storage_id).reset_abuse(channel_id)
+    rescue ArgumentError, OpenSSL::Cipher::CipherError
+      raise ActionController::BadRequest.new, "Bad channel_id"
+    end
+    render json: {abuse_score: value}
+  end
+
+  # PATCH /v3/(animations|assets|sources|files|libraries)/:channel_id?abuse_score=:abuse_score
+  def update_file_abuse
+    return head :unauthorized unless can?(:update_file_abuse, nil)
+
+    value = update_file_abuse_score(params[:endpoint], params[:encrypted_channel_id], params[:abuse_score])
+
+    render json: {abuse_score: value}
+  end
+
+  # Non-actions, public methods so they can be tested easier.
+  # The methods below are in this controller because they depend on the
+  # storage_id helper, which has dependencies on current_user.
+
+  # Increment an abuse score.
+  # Was POST /v3/channels/:channel_id/abuse
+  def update_channel_abuse_score(channel_id)
+    # Reports of abuse from verified teachers are more reliable than reports
+    # from students so we increase the abuse score enough to block the project
+    # with only one report from a verified teacher.
+    #
+    # Temporarily ignore anonymous reports and only allow verified teachers
+    # and signed in users to report.
+    restrict_reporting_to_verified_users = DCDO.get('restrict-abuse-reporting-to-verified', true)
+    amount =
+      if current_user&.verified_teacher?
+        20
+      elsif current_user && !restrict_reporting_to_verified_users
+        10
+      else
+        0
+      end
+    begin
+      value = Projects.new(get_storage_id).increment_abuse(channel_id, amount)
+    rescue ArgumentError, OpenSSL::Cipher::CipherError
+      raise ActionController::BadRequest.new, "Bad channel_id"
+    end
+    value
+  end
+
+  # Update all assets for the given channelId to have the provided abuse score
+  # Was PATCH /v3/(animations|assets|sources|files|libraries)/:channel_id?abuse_score=<abuse_score>
+  def update_file_abuse_score(endpoint, encrypted_channel_id, abuse_score)
+    return if abuse_score.nil?
+
+    buckets = get_bucket_impl(endpoint).new
+
+    begin
+      files = buckets.list(encrypted_channel_id)
+    rescue ArgumentError, OpenSSL::Cipher::CipherError
+      raise ActionController::BadRequest.new, "Bad channel_id"
+    end
+    files.each do |file|
+      break unless can_update_abuse_score?(endpoint, encrypted_channel_id, file[:filename], abuse_score)
+      buckets.replace_abuse_score(encrypted_channel_id, file[:filename], abuse_score)
+    end
+
+    abuse_score
+  end
+
+  private
+
+  def project_validator?
+    return false unless current_user
+    current_user.project_validator?
```

Comment:
nit:
```suggestion
    current_user&.project_validator?
```
at which point you could also just inline this method